pkg/proxy/namespace/namespace.go

```diff
@@ -0,0 +1,14 @@
+package namespace
+
+import "net/url"
+
+// Namespace represents a namespace in a member cluster for the specific sso user-workspace
+type Namespace struct {
+	Username    string
+	ClusterName string
+	APIURL      url.URL
+	Namespace   string
+	Workspace   string
+	// Target cluster token
+	TargetClusterToken string
```

Comment:
what does `TargetClusterToken` mean exactly? Or can we rename it to `Token` instead?
Also, the struct is called `Namespace` but it has a `Username` field and a `Namespace` field, so should we rename this struct to `UserNamespace` to make it explicit that it's for a specific User in a specific Namespace?

Reply:
I added a better comment to the `TargetClusterToken` field.
Regarding the `UserNamespace` vs `Namespace`, this structs actually represents the information needed for accessing specific namespace for specific user in the specific workspace. So, there are three dimensions here: Namespace, User, Workspace. It's so because user can have access to different workspaces.
And while I'm agree that `Namespace` is not perfect I don't thing that UserNamespace will be much better.

Reply:
> Also the comment of the type - because it doesn't represent the namespace at all. It doesn't have any name of the namespace stored anywhere, but correct me if I'm wrong.

It did have the namespace name. Initially the struct had all the information about the namespace in case we need it in the future. The purpose of this struct was more general then just the SA token. But Xavier asked me to remove it because it's not use yet :)
So, it's really about an access to specific namespace for a specific user for a specific workspace.
So, maybe I will rename it to ClusterAccess but I'm not super happy with this name either...

Reply:
I've renamed it to "NamespaceAccess" in https://github.com/codeready-toolchain/registration-service/pull/224/commits/b5e7f4e8e266901c0d91394b184396abf38664da

Reply:
`NamespaceAccess` sounds good as well. Thanks :+1: